test/export/InstanceExporter.test.ts

```diff
@@ -583,6 +583,264 @@ describe('InstanceExporter', () => {
+      );
+      testInstance.rules.push(valueQuantityAssignment);
+      doc.instances.set(testInstance.name, testInstance);
+
+      const exported = exportInstance(testInstance);
+      expect(exported.toJSON()).toEqual({
+        resourceType: 'Observation',
+        id: 'TestInstance',
+        meta: {
+          profile: ['http://hl7.org/fhir/us/minimal/StructureDefinition/ObservationProfile']
+        },
+        status: 'final',
+        code: { coding: [{ code: 'testcode' }] },
+        valueQuantity: {
+          id: 'some-id',
+          value: 5,
+          system: 'http://unitsofmeasure.org',
+          code: 'mm'
+        }
+      });
+    });
+
+    it('should assign fixed values from value[x] children using the correct specific choice property name (primitive edition)', () => {
+      // Profile: ObservationProfile
+      // Parent: Observation
+      const observationProfile = new Profile('ObservationProfile');
+      observationProfile.parent = 'Observation';
+      // * value[x] 1..1
+      const valueCardRequired = new CardRule('value[x]');
+      valueCardRequired.min = 1;
+      valueCardRequired.max = '1';
+      observationProfile.rules.push(valueCardRequired);
+      // * value[x].id 1..1
+      const valueIdCardRequired = new CardRule('value[x].id');
+      valueIdCardRequired.min = 1;
+      valueIdCardRequired.max = '1';
+      observationProfile.rules.push(valueIdCardRequired);
+      // * value[x].id = "some-required-id"
+      const valueIdAssignment = new AssignmentRule('value[x].id');
+      valueIdAssignment.value = 'some-id';
+      observationProfile.rules.push(valueIdAssignment);
+      doc.profiles.set(observationProfile.name, observationProfile);
+
+      // Instance: TestInstance
+      // InstanceOf: ObservationProfile
+      const testInstance = new Instance('TestInstance');
+      testInstance.instanceOf = 'ObservationProfile';
+      // * status = #final
+      const statusFinal = new AssignmentRule('status');
+      statusFinal.value = new FshCode('final');
+      testInstance.rules.push(statusFinal);
+      // * code = #testcode
+      const codeTestCode = new AssignmentRule('code');
+      codeTestCode.value = new FshCode('testcode');
+      testInstance.rules.push(codeTestCode);
+      // * valueString = 'Hello World'
+      const valueStringAssignment = new AssignmentRule('valueString');
+      valueStringAssignment.value = 'Hello World';
+      testInstance.rules.push(valueStringAssignment);
+      doc.instances.set(testInstance.name, testInstance);
+
+      const exported = exportInstance(testInstance);
+      expect(exported.toJSON()).toEqual({
+        resourceType: 'Observation',
+        id: 'TestInstance',
+        meta: {
+          profile: ['http://hl7.org/fhir/us/minimal/StructureDefinition/ObservationProfile']
+        },
+        status: 'final',
+        code: { coding: [{ code: 'testcode' }] },
+        valueString: 'Hello World',
+        _valueString: { id: 'some-id' }
```

Comment:
I did notice that this `id` got set properly in the output, but I also got errors telling me that the `id` was missing. I'm not sure if this is part of this PR or separate. These are the two errors: 

```
Element Observation.value[x].id has minimum cardinality 1 but occurs 0 time(s).
Element Observation.value[x]:valueString.id has minimum cardinality 1 but occurs 0 time(s)
```

Reply:
@cmoesel sorry I noticed those after I clicked the approve button. Let me know if you think if it's separate.

Reply:
Oh dang.  I wasn't checking the errors in the tests I guess.  Ugh.

That actually does seem like a separate issue though.  It has nothing to do with the bug I'm attempting to fix here, but rather is a more general issue w/ validation of children of choice properties.  So... I'd be inclined to say it's a separate bug...  What do you think?

Reply:
I think I'm inclined to say it's separate. It does introduce a new error, but I'm guessing there hasn't been an error because we've had the incorrect 

```
 "value[x]": {
    "id": "some-required-id"
  },
```
on the instance. So I think correct output + a new error > incorrect output - error. Also, on the surface (because I haven't truly looked to see where it comes from), the error does seem separate from the fixes here. So I'm good with leaving it separate.